File: Go/Algorithms/selectionSort.go
package main

func selectionSort(array []int) []int {
	for i := 0; i < len(array); i++ {
		var min int = i;

		for j := i+1; j < len(array); j++ {
			if array[min] > array[j] {
				min = j;
			}
		}
		if min != i {
			temp := array[i];
			array[i] = array[min];
			array[min] = temp;
		}
	}
	return array;
}

File: Nim/Algorithms/selection_sort.nim
proc selection_sort(arr: var seq[int]): seq[int] = 
    for i in 0..<arr.len:
        var min: int = i

        for j in i+1..<arr.len:
            if arr[min] > arr[j]:
                min = j
        
        if min != i:
            var temp: int = arr[i]
            arr[i] = arr[min]
            arr[min] = temp
    
    return arr